### PR TITLE
Chore/frontend structure

### PR DESCRIPTION
**Pull request by frontend-specialisten**
- [x] Document new naming conventions / folder structure
- [x] Move libs/dh-app -> libs/dh
- [x] Move apps/ett-app -> apps/ett/app-ett
- [x] Move apps/dh-app -> apps/dh/dh-ett 
- [x] Move apps/ett-app-e2e -> apps/ett/e2e-ett
- [x] Move apps/dh-app-e2e -> apps/dh/e2e-dh 
- [x] Rename dh-app -> app-dh
- [x] Rename ett-app -> app-ett
- [x] Move BFF from libs/api-dh-app -> apps/dh/api-dh
- [x] [Add documentation of "false positives" of merging frontend pull-requests](https://github.com/Energinet-DataHub/MightyDucks/issues/231)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,20 +8,139 @@ Monorepo that hosts:
...
+├── dist                    # Contains output files when building artifacts (for distribution)
+|  ├── apps                 #
+|  └── libs                 #
+├── libs                    # Contains source code for libraries. Libraries are grouped by a product root folder, and type prefixed sub-folder(s).
+|  ├── dh                   # - Libraries for Data Hub (product)
+|  ├── ett                  # - Libraries for Energy Track and Trace (product)
+|  └── ui-watt              # - Actual implementation of the design system
+└── tools                   # Contains various tools
+   └── executors            # - Executors perform actions on your code. This can include building, linting, testing, serving. Example: "add license"
+```
```
Comment (LayZeeDK) [inline README.md:34]: praise: awesome tree

File: README.md [hunk 2]
```diff
@@ -8,20 +8,139 @@ Monorepo that hosts:
...
+|  |  ├── app-dh            # - Frontend for Data Hub
+|  |  └── e2e-dh            # - E2E tests for Data Hub
+|  └── ett                  # Energy Track and Trace (product)
+|  |  ├── app-ett           # - Frontend for Energy Track and Trace
+|  |  └── e2e-ett           # - E2E tests for Energy Track and Trace
+├── dist                    # Contains output files when building artifacts (for distribution)
+|  ├── apps                 #
+|  └── libs                 #
+├── libs                    # Contains source code for libraries. Libraries are grouped by a product root folder, and type prefixed sub-folder(s).
+|  ├── dh                   # - Libraries for Data Hub (product)
```
Comment (LayZeeDK) [inline README.md:29]: suggestion: add `gf` product folder
```suggestion
|  ├── dh                   # - Libraries for Data Hub (product)
|  ├── gf                   # - Shared libraries for GreenForce (ART)
```

File: README.md [hunk 3]
```diff
@@ -8,20 +8,139 @@ Monorepo that hosts:
...
+|  |  ├── api-dh            # - BFF for Data Hub
+|  |  ├── app-dh            # - Frontend for Data Hub
+|  |  └── e2e-dh            # - E2E tests for Data Hub
+|  └── ett                  # Energy Track and Trace (product)
+|  |  ├── app-ett           # - Frontend for Energy Track and Trace
+|  |  └── e2e-ett           # - E2E tests for Energy Track and Trace
+├── dist                    # Contains output files when building artifacts (for distribution)
+|  ├── apps                 #
+|  └── libs                 #
+├── libs                    # Contains source code for libraries. Libraries are grouped by a product root folder, and type prefixed sub-folder(s).
```
Comment (LayZeeDK) [inline README.md:28]: suggestion: mention domain grouping folders

File: README.md [hunk 4]
```diff
@@ -8,20 +8,139 @@ Monorepo that hosts:
...
+|  ├── dh                   # - Libraries for Data Hub (product)
+|  ├── ett                  # - Libraries for Energy Track and Trace (product)
+|  └── ui-watt              # - Actual implementation of the design system
+└── tools                   # Contains various tools
+   └── executors            # - Executors perform actions on your code. This can include building, linting, testing, serving. Example: "add license"
```
Comment (LayZeeDK) [inline README.md:33]: suggestion: list the `generators` folder

File: README.md [hunk 5]
```diff
@@ -8,20 +8,139 @@ Monorepo that hosts:
 - CI/CD pipelines
 - Infrastructure
 
+## Folder Structure
+
```
Comment (LayZeeDK) [inline README.md:12]: suggestion: mention the `docs` and `infrastructure` folders

File: README.md [hunk 6]
```diff
@@ -8,20 +8,139 @@ Monorepo that hosts:
...
+A E2E app contains the e2e application of a product. Frontend apps are located under `apps/<product>` and are prefixed with `e2e-<product>`
+Allowed dependencies: `e2e-util` libraries and always an implicit dependency of an app, and a api (BFF).
+
+## Library Types
+
+All libraries are related to a product or shared across multiple products. The libraries are located under `libs/<product>` and are prefixed with `<library type>-<library name>`.
+For shared libraries `gf` should be used as `product`.
+
+No library must have any dependencies to an app. Depending on the type of a library, a library can have dependencies to other certain types of libraries.
```
Comment (LayZeeDK) [inline README.md:61]: suggestion: `data-access` libraries may depend on `api` apps

**Review by LayZeeDK — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (frontend-specialisten) [reply README.md]: Should be fixed now.

File: README.md [hunk 7]
```diff
@@ -8,20 +8,151 @@ Monorepo that hosts:
...
+|  |  └── e2e-dh            # - E2E tests for Data Hub
+|  └── ett                  # Energy Track and Trace (product)
+|  |  ├── app-ett           # - Frontend for Energy Track and Trace
+|  |  └── e2e-ett           # - E2E tests for Energy Track and Trace
+├── build                   # Contains infrastructure for Data hub and the design system
+├── dist                    # Contains output files when building artifacts (for distribution)
+├── docs                    # Contains general documentation
+├── dist                    # Contains output files when building artifacts (for distribution)
+|  ├── apps                 #
+|  └── libs                 #
```
Comment (dzhavat) [inline README.md:30]: nitpick: "dist" is mentioned twice

File: README.md [hunk 8]
```diff
@@ -8,20 +8,151 @@ Monorepo that hosts:
...
+         └── <library type>-<library name>   # A domain can contain serveral libraries which are prefixed by type
+```
+
+## App Types
+
+- api (BFF)
+- app (Frontend)
+- e2e
+
+### Api apps (BFF's)
```
Comment (dzhavat) [inline README.md:54]: nitpick:

```suggestion
### API apps (BFFs)
```

File: README.md [hunk 9]
```diff
@@ -8,20 +8,151 @@ Monorepo that hosts:
...
+
+A Api app contains the BFF of a product. Api apps are located under `apps/<product>` and are prefixed with `api-<product>`.
```
Comment (dzhavat) [inline README.md:56]: nitpick:

```suggestion
An API app contains the BFF of a product. API apps are located under `apps/<product>` and are prefixed with `api-<product>`.
```

File: README.md [hunk 10]
```diff
@@ -8,20 +8,151 @@ Monorepo that hosts:
...
+
+### Frontend App
```
Comment (dzhavat) [inline README.md:58]: nitpick:

```suggestion
### Frontend Apps
```

File: README.md [hunk 11]
```diff
@@ -8,20 +8,151 @@ Monorepo that hosts:
...
+
+A Frontend app contains the frontend application of a product. Frontend apps are located under `apps/<product>` and are prefixed with `app-<product>`.
+Allowed dependencies: `shell`, `environments`, `assets`, `styles` libraries.
+
+### E2E apps
+
+A E2E app contains the e2e application of a product. Frontend apps are located under `apps/<product>` and are prefixed with `e2e-<product>`
+Allowed dependencies: `e2e-util` libraries and always an implicit dependency of an app, and a api (BFF).
```
Comment (dzhavat) [inline README.md:66]:
```suggestion
An E2E app contains the e2e application of a product. E2E apps are located under `apps/<product>` and are prefixed with `e2e-<product>`
Allowed dependencies: `e2e-util` libraries and always an implicit dependency on an app, and an API (BFF).
```

File: README.md [hunk 12]
```diff
@@ -8,20 +8,151 @@ Monorepo that hosts:
...
+
+## Library Types
+
+All libraries are related to a product or shared across multiple products. The libraries are located under `libs/<product>` and are prefixed with `<library type>-<library name>`.
+For shared libraries `gf` should be used as `product`.
+
+Only libraries of type `data-access` may have dependencies to apps, and only apps of type `api`. Depending on the type of a library, a library can have dependencies to other certain types of libraries.
```
Comment (dzhavat) [inline README.md:73]:
```suggestion
Only libraries of type `data-access` may have dependencies to apps, and only apps of type `api`. Depending on the type of a library, a library can have dependencies to other types of libraries:
```

File: README.md [hunk 13]
```diff
@@ -8,20 +8,151 @@ Monorepo that hosts:
...
+Allowed dependencies to: `data-access`, `util`, `test-util`, `domain`.
+
+### e2e-util
+
+Cypress commands or fixtures
+Allowed dependencies to: `util`, `test-util`, `e2e-util`.
+
+### domain
+
+interfaces, types, constants, functions and services related to domain objects.
```
Comment (dzhavat) [inline README.md:119]:
```suggestion
Interfaces, types, constants, functions and services related to domain objects.
```

File: README.md [hunk 14]
```diff
@@ -8,20 +8,151 @@ Monorepo that hosts:
...
+Allowed dependencies to: `feature`, `ui`, `data-access`, `util`, `test-util`, `shell`, `environments`, `assets`, `styles`.
+
+### environments
+
+Angular environment files.
+Allowed dependencies to: `util`, `test-util`, `environments`.
+
+### assets
+
+Icons, images, fonts etc.
```
Comment (dzhavat) [inline README.md:134]:
```suggestion
Icons, images, fonts, etc.
```

**Review by dzhavat — CHANGES_REQUESTED** [review]

File: README.md [hunk 15]
```diff
@@ -31,7 +161,9 @@ There's currenly only one BFF located in `api-dh-app` under `libs`. It is for `d
 
 Located under `.github/workflows`. There are:
 
-- `api-dh-app-ci.yml` - Used by the BFF for `dh-app`.
+- `api-dh-ci.yml` - Used by the BFF for `app-dh`.
 - `ett.yml` - Used by "Energy Track and Trace" app
 - `license-check-ci.yml` - Used for adding license to files
 - `workspace.yml` - Used to build, test, format and lint all front-end apps
+
+**Please note**: Before merging a frontend pull request, be sure that all checks have passed. Currently, jobs done by a Git-bot (like formatting and adding licenses) don't trigger a re-run of all the other checks. The "merge button" can therefore give you a false positive signal.
```
Comment (dzhavat) [inline README.md:169]: question: Do you think we should add this to the Pull Request template as well? So the chance of seeing it is higher. Putting it here makes it a bit hidden.

**Comment by LayZeeDK** [comment]
We should add an Nx `project.json` configuration file to `api-dh` and set up implicit dependencies between DataHub API data access libraries and DataHub end-to-end tests.

Something like
```json
{
  "projectType": "application",
  "root": "apps/dh/api-dh",
  "sourceRoot": "apps/dh/api-dh/source",
  "prefix": "dh",
  "targets": {},
  "tags": ["product:dh", "type:api"]
}
```

File: README.md (hunk 15 above)
Comment (frontend-specialisten) [reply README.md:169]: Good point, should be done now

File: .github/workflows/ett.yml [hunk 16]
```diff
@@ -1,7 +1,7 @@
 name: Energy Track and Trace
 
 env:
-  app-name: ett-app
+  app-name: app-ett
```
Comment (LayZeeDK) [inline .github/workflows/ett.yml:4]: issue(blocking): when this is changed, it affects the Docker image name in Line 50 which is used for deployment, effectively breaking ETT's infrastructure instantly

**Review by LayZeeDK — CHANGES_REQUESTED** [review]

File: .github/workflows/ett.yml (hunk 16 above)
Comment (frontend-specialisten) [reply .github/workflows/ett.yml:4]: We should properly talk about this then, I did change the path of the dist in the docker file but if there's more depending on the app-name which is not part of the repo I need some help.

File: .github/workflows/ett.yml [hunk 17]
```diff
@@ -138,13 +138,13 @@ jobs:
...
           name: ${{ env.app-name }}
-          path: ./dist/apps/${{ env.app-name }}/
+          path: ./dist/apps/ett/${{ env.app-name }}/
 
       - name: Container image name
         id: image-name
         uses: ASzc/change-string-case-action@v2
         with:
-          string: ghcr.io/${{ github.repository }}-${{ env.app-name }}
+          string: ghcr.io/${{ github.repository }}-ett-app
```
Comment (LayZeeDK) [inline .github/workflows/ett.yml:147]: suggestion: let's add a comment for me to solve
```suggestion
          # TODO(xlgni): coordinate name change to `env.app-name` with Fenris infrastructure
          string: ghcr.io/${{ github.repository }}-ett-app
```

Comment (frontend-specialisten) [reply .github/workflows/ett.yml:148]: Good point! :D

**Review by LayZeeDK — CHANGES_REQUESTED** [review]

**Review by LayZeeDK — APPROVED** [review]

**Pull request merged**